### DISCUSSION TITLE
command shadowed by xonsh function name

### DISCUSSION THREAD

**Discussion by tkossak** [xonsh/xonsh#5428] [Q&A] [answered]
I have cli app and xonsh function with the same name: `tabulate`.
When I write it in terminal, xonsh executes the function. Is there any escape mechanism that lets me run cli app instead of function?
I know I can just write full path eg `/usr/local/bin/tabulate`, but maybe there's something shorter/quicker?

**Comment by anki-code** [comment, marked as answer]
Hey! First of all there is no builtin `tabulate` in xonsh:
```xsh
xonsh --no-rc --no-env  # `--no-env` is from xonsh > 0.16.0
tabulate
# command not found: 'tabulate'
```
If we emulate this case:
```xsh
tabulate = lambda: 0

cd /tmp
echo '#!/bin/bash' > tabulate
chmod +x tabulate
$PATH.append($PWD)
which tabulate
# /tmp/tabulate

tabulate
# <function __main__.<lambda>>
```
Workaround 1:
```xsh
del tabulate  # just remove function
tabulate
# success
```
Workaround 2:
```xsh
@('tabulate')  # use command substitution in subprocess mode
# success
```
Workaround 3:
```xsh
aliases['tab'] = $(which tabulate)  # create alias (add `.rstrip()` if xonsh <0.17.0)
tab
# success
```
Workaround 4 - full or partial uppercase:
```xsh
TABULATE
# success
Tabulate
# success
```

Take a look into [xonsh-cheatsheet](https://github.com/anki-code/xonsh-cheatsheet/).

**Comment by tkossak** [comment]
Thank you!
Btw. i see this option `--no-env` in docs and in your post, but my xonsh (v0.16.0) doesn't have this option. Was it removed or something is wrong with my configuration?

```
$ ~ @ xonsh --no-env
usage: xonsh [-h] [-V] [-c COMMAND] [-i] [-l] [--rc RC [RC ...]] [--no-rc] [--no-script-cache]
             [--cache-everything] [-D ITEM] [--shell-type SHELL_TYPE] [--timings]
             [script-file] ...
xonsh: error: unrecognized arguments: --no-env
```

**Reply by anki-code** [reply]
Ah, sorry, It will be released in future version of xonsh (https://github.com/xonsh/xonsh/pull/5370) but it's really useful already now ;)
You can install xonsh from the main branch to have it:
```xsh
xpip install git+https://github.com/xonsh/xonsh
```
